demo/components/table/table-demo.ts

```diff
@@ -24,79 +31,116 @@ export class TableDemo implements OnInit {
     {title: 'Start date', name: 'startDate'},
     {title: 'Salary', name: 'salary'}
   ];
-  public page:number = 1;
-  public itemsPerPage:number = 10;
-  public maxSize:number = 5;
-  public numPages:number = 1;
-  public length:number = 0;
-
-  public config:any = {
-    paging: true,
+
+  config:any = {
     sorting: {columns: []},
-    filtering: {filterString: '', columnName: 'position'}
+    filtering: {filterString: '', columnName: 'position'},
+    pagination: {
+      page: 1,
+      itemsPerPage: 10,
+      maxSize: 5,
+      numPages: 1,
+    }
   };
 
   private data:Array<any> = TableData;
 
-  constructor() {
-    this.length = this.data.length;
+  ngOnInit() {
+    let initialTableComponentsState = [
```

Comment:
@rychkog chain it please (initialTableComponentsState, tableEventsStream, filteredRows, paginatedRows, sortedRows)